go/src/koding/klient/machine/machinegroup/id.go

```diff
@@ -24,22 +26,36 @@ type IDResponse struct {
 // aliases and machine IP addresses. Supported identifiers are:
 //  - machine full ID string.
 //  - machine alias.
+//  - machine label in form [owner@]label.
 //  - machine IP address.
 func (g *Group) ID(req *IDRequest) (*IDResponse, error) {
 	if req == nil {
 		return nil, errors.New("invalid nil request")
 	}
 
+	owner, identifier := "", req.Identifier
+	if toks := strings.SplitN(identifier, "@", 2); len(toks) > 1 {
```

Comment:
just curious:

do we allow `@` in identifier?

asking because toks[1] might have more `@`s in it

Reply:
I believe we intentionally treat tok[1] (`label`) as an arbitrary string, so it can have any characters. Later lookup will just fail with 404.

Reply:
this was discussed [here](https://github.com/koding/koding/issues/10961#issuecomment-292802212).

It doesn't cover all cases eg. `unix@machine` label which will fail due to `unix` token parsed as username. However, we still are able to workaround this by `@unix@machine`.

Reply:
why not searching for all occurrences and ask for the which one if there are multiple? So, you have a `machine` and tried to use `foo@machine`, search for `machine` and `foo@machine` respectively; if only one of them found use it as is, if there are multiple result ask for what to do. like in here https://github.com/koding/koding/issues/10961#issuecomment-298421459 `@unix@machine` is completely weird 👎

Reply:
@gokmen the problem with asking is that we cannot use this approach in non-interactive kd calls (eg atom IDE calls `kd mount <labal>`).